src/StreamingDataFrames/streamingdataframes/models/serializers/quix.py

```diff
@@ -132,91 +213,96 @@ def __call__(self, value: bytes, ctx: SerializationContext) -> Iterable[dict]:
+            QModelKey.HEADER_NAME: value.pop(LegacyHeaders.Q_MODEL_KEY),
         }
-    """
+        message = value.pop("V")
+        return headers, message
 
-    def deserialize(
-        self, model_key: str, value: Union[List[Mapping], Mapping]
-    ) -> Iterable[Mapping]:
-        # Fail if model key is unsupported
-        if model_key not in (QModelKey.TIMESERIESDATA, QModelKey.PARAMETERDATA):
-            raise SerializationError(
-                f'"{QModelKey.HEADER_NAME}" must be "{QModelKey.TIMESERIESDATA}" '
-                f'or {QModelKey.PARAMETERDATA}, got"{model_key}"'
-            )
-        if not isinstance(value, Mapping):
-            raise SerializationError(f'Expected mapping, got type "{type(value)}"')
 
-        timestamps = value["Timestamps"]
-        # Make a list of parameters and iterators with values to get them one by one
-        # and decode values from base64 if they're binary
-        all_params = [
-            (
-                param_name,
-                iter(param_values),
-                True if param_type == "BinaryValues" else False,
-            )
-            for param_type in ("NumericValues", "StringValues", "BinaryValues")
-            for param_name, param_values in value.get(param_type, {}).items()
-        ]
-        # Do the same with TagValues
-        tags = [
-            (tag, iter(values)) for tag, values in value.get("TagValues", {}).items()
-        ]
+class QuixLegacySerializer(JSONSerializer):
+    extra_headers = {}
+    _legacy = {}
 
-        # Iterate over timestamps and get a corresponding parameter value
-        # for each timestamp
-        for timestamp_ns in timestamps:
-            row_value = {
-                param: _b64_decode_or_none(next(values)) if is_binary else next(values)
-                for param, values, is_binary in all_params
+    def __init__(
+        self,
+        as_legacy: Optional[bool] = None,
+        dumps: Optional[Callable[[Any, None], Union[str, bytes]]] = None,
+        dumps_kwargs: Optional[Mapping] = None,
+    ):
+        """
+        Serializer that returns data in json format.
+        :param as_legacy: parse as the legacy format; Default = True
+        :param dumps: a function to serialize objects to json. Default - `json.dumps`
+        :param dumps_kwargs: a dict with keyword arguments for `dumps()` function.
+        """
+        if as_legacy is None:
+            as_legacy = True
+        self.as_legacy = as_legacy
+        super().__init__(dumps=dumps, dumps_kwargs=dumps_kwargs)
+        if self.as_legacy:
+            self._legacy = {
+                QModelKey.HEADER_NAME: QModelKey.PARAMETERDATA,
+                QCodecId.HEADER_NAME: QCodecId.JSON_TYPED,
+            }
+        else:
+            self.extra_headers = {
+                QModelKey.HEADER_NAME: QModelKey.TIMESERIESDATA,
+                QCodecId.HEADER_NAME: QCodecId.JSON_TYPED,
             }
```

Comment:
Could you please clarify how does this part play with the descendant classes?

I see it changes `self._as_legacy` here, but the descendants also [change it](https://github.com/quixio/quix-streams/pull/208/files#diff-432db6a0e20f9f6a1b464971827eef744d306f48c8adcc1b0b3a7b207c0b8de3R345-R352)


Looks like this base class can be replaced by a simple function `_as_quix_legacy(value: bytes, model_key: str,  codec_id: str)` that will wrap the serialized JSON into some envelope

Reply:
I have cleaned up the approach (and removed stuff that I thought I cleaned up, sorry, I should have double-checked)

While I probably could remove the inheritance, please take a look at my latest commit and see if you are happy with my improved approach. 

I still have a `QuixSerializer` base class for the sake of not repeating stuff across both the `Timeseries` and `EventData` serializers.

Also, just in case you were wondering what I was doing with the headers, I am basically making sure I do not include the `extra_headers` if it's in the legacy format (in case that wasn't clear).